launchable/commands/record/commit.py

```diff
@@ -10,11 +19,67 @@
               default=os.getcwd(),
               type=str
               )
-def commit(source):
-    token, org, workspace = parse_token()
+@click.option('--executable',
+              help="collect commits with Jar or Docker",
+              type=click.Choice(['jar', 'docker']),
+              default='jar'
+              )
+def commit(source, executable):
     source = os.path.abspath(source)
+
+    if executable == 'jar':
+        download_jar()
+        exec_jar(source)
+    elif executable == 'docker':
+        exec_docker(source)
+
+
+def download_jar():
+    pathlib.Path(jar_dir_path).mkdir(parents=True, exist_ok=True)
+
+    res = requests.get(ingester_jar_url, stream=True)
+    total_length = int(res.headers.get('content-length'))
+
+    if os.path.exists(jar_file_path) and os.path.getsize(jar_file_path) == total_length:
+        print("exe_deploy.jar exists in {}".format(jar_dir_path))
+    else:
+        total_downloaded = 0
+        f = open(jar_file_path, 'wb')
+        for chunk in res.iter_content(chunk_size=512 * 1024):
+            if chunk:
+                f.write(chunk)
+
+                total_downloaded += len(chunk)
+                sys.stdout.write("\rDownloading exe_deploy.jar: {:.1f}% ({:,} / {:,} byte)".format(
```

Comment:
For interactive use this status report is nice but for CI systems where this command is expected to be used, this is effectively no different from NL, so it'd be quite noisy.

Not a major point, but I'd suggest toning this down. For example, one line to report the start of download (some CI systems can only show a line after it sees a whole line), then one dot for every 10MB, and NL at the end of download, would be nice. Alternatively, you can check if the stdout is a terminal, and use that as a cue to produce output geared for humans like this.

(But really, the whole on demand download is an extra nice-to-have, so it's probably not wise to get too smart in that regard.)

Reply:
Okay. I fixed to tone down this prompt like this
```
Downloading exe_deploy.jar (262.03 MB)
....
```